rfc/new-artifacts-service/README.md

```diff
@@ -0,0 +1,207 @@
+# Separate artifacts to new service
+
+| | |
+|-|-|
+| **Status**     | **Proposed**, Accepted, Implemented, Obsolete |
+| **RFC #**      | _Update after PR has been made_ |
+| **Author(s)**  | [Nemesis Osorio](https://github.com/nemesisOsorio), [Sergio Quintero](https://github.com/sergio-quintero) |
+| **SIG / WG**   | _Applicable SIG(s) or Working Group(s)_ |
+
+## Overview
+
+Users with large context pipelines do not have a good experience with pipeline executions. <br/>
+Context and outputs in pipeline executions can contain large artifacts/objects such as k8s manifest (yaml/json files). <br/>
+Spinnaker passes these artifacts/objects in request and response bodies across some services. <br/>
+The idea is to create a new service to store these artifacts/objects
+and only pass the `id`s of those artifacts/objects in the request/response bodies and context/outputs.
+
+Deck users do not click to see the produced artifact/object in every single pipeline execution.
+With this approach, artifacts will be loaded on demand (lazy loading).
+
+### Goals and Non-Goals
+
+#### Goals
+
+* Improve Deck performance with large pipeline executions.
+* Optimize message size over the wire.
+
+#### Non-Goals
+
+* Move [clouddriver-artifacts](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-artifacts)
+to the new service
+* Refactor Angular code in Deck.
+* Add new permissions to Fiat service.
+* Avoid the duplication of artifacts when executing a pipeline multiple times without changes.
+
+## Motivation and Rationale
+
+* Deck is very slow and sometimes freezes when you are producing/consuming large artifacts/objects in pipeline executions.
+* Smaller payloads are sent in request/response.
+* Save serializing and deserializing computations.
+
+Early customers:
+* Armory on behalf of its customers.
+
+## Timeline
+
+The proposed plan is to implement this by pipeline stages:
+
+* Milestone 1: (Targeting 1.32) Address stages with the most acute performance issues
+  * Introduce new Artifact service
```

Comment:
A big one not answered here is the installation of a new service, config and how that'd be done.

Reply:
We prefer to avoid changes in Halyard, do you think raw yml manifests might be the way?

We will definitely add a way to deploy using Operator.